debezium-embedded/src/main/java/io/debezium/engine/source/EngineSourceTask.java

```diff
@@ -31,4 +35,11 @@ public DebeziumSourceTaskContext context() {
     public SourceTask connectTask() {
         return connectTask;
     }
-}
\ No newline at end of file
+
+    @SuppressWarnings("unchecked")
+    public Optional<BaseSourceTask<?, ?>> debeziumConnectTask() {
```

Comment:
That's true only for our tasks.  The entire implementation still seams to count with the possibility of running 3rd party connectors. Signalling obviously won't work there but this ensure a no-op instead of failure should anyone attempt to use it.

Reply:
I don't think any task running inside Debezium server doesn't inherit from `BaseSourceTask`, but why not to check it explicitly. I'd just rename it to something like `debeziumSourceTask` or something, just to avoid some connotation to Kafka Connect.

Reply:
Any taks that will not be from Debezium connector will not inherit from BaseSourceTask (as that is OUR class).

Reply:
Do you envision Debezium server will be used one day for something else then running Debezium tasks? As I wrote, I have no problem with it, we should have ambitions!

Reply:
On several occasions it was publicaly stated that it does. And it was used to do that in the past by some people. I'm not saying it should be support, I just didn't want to break it until we decide to do so.